third_party/terraform/tests/resource_spanner_database_test.go

```diff
@@ -81,3 +138,21 @@ func TestDatabaseNameForApi(t *testing.T) {
 	expected := "projects/project123/instances/instance456/databases/db789"
 	expectEquals(t, expected, actual)
 }
+
+// Unit Tests for ForceNew when the change in ddl
+func TestSpannerDatabase_resourceSpannerDBDdlCustomDiffFunc(t *testing.T) {
+	t.Parallel()
+
+	d := &ResourceDiffMock{
+		Before: map[string]interface{}{
+			"ddl": []interface{}{"CREATE TABLE t1 (t1 INT64 NOT NULL,) PRIMARY KEY(t1)"},
+		},
+		After: map[string]interface{}{
+			"ddl": []interface{}{"CREATE TABLE t2 (t2 INT64 NOT NULL,) PRIMARY KEY(t2)"},
+		},
+	}
+	err := resourceSpannerDBDdlCustomDiffFunc(d)
+	if err != nil {
+		t.Errorf("failed, expected no error but received one - %s", err)
```

Comment:
This is a good start! I think in addition to checking whether or not there was an error, you should also check whether or not the diff was marked as ForceNew. I can imagine a few test cases:
1. What you have now (basic forcenew check)
2. Before and after match (basic non-forcenew check)
3. A few statements where just the final one is different (should be non-forcenew)
4. A few statements where they match but are in a different order (should be forcenew)
5. A few statements where the before contains one more statement than the after (should be forcenew)